homeassistant/components/linky/__init__.py

```diff
@@ -1 +1,57 @@
 """The linky component."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DEFAULT_TIMEOUT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up Linky sensor from legacy config file."""
+    _LOGGER.error("LINKY_INIT:async_setup")
+
+    _LOGGER.error(config["sensor"])
+    sensors_conf = config["sensor"]
```

Comment:
We can access the config section of this domain with the domain key.
```suggestion
    conf = config[DOMAIN]
```

Reply:
Nope, it's not working like this.

With `conf = config[DOMAIN]` and selecting the `for` after that : 
<img width="873" alt="Capture d’écran 2019-08-28 à 13 36 02" src="https://user-images.githubusercontent.com/14821482/63852647-15892700-c999-11e9-8ea7-dec52180f777.png">
So "unable to setup the platform"

With `sensors_conf = config["sensor"]` and the `for` : 
<img width="863" alt="Capture d’écran 2019-08-28 à 13 36 51" src="https://user-images.githubusercontent.com/14821482/63852727-44070200-c999-11e9-9544-072f9417bcaf.png">
And working


Don't forget that the original configuration was : 
```yaml
  - platform: linky
    username: __email__
    password: __password__
```

and not 
```yaml
linky:
    username: __email__
    password: __password__
```

Reply:
The configuration should move under the linky key in the configuration.yaml.

Reply:
Okey, that's also why you added the comment just below.

I thought that the config.yml will become deprecate since the config flow is prefered now, so no needed to modify the existing conf.

Will do it :+1: